rust/lance/src/dataset.rs

```diff
@@ -1466,6 +1436,56 @@ impl Dataset {
     }
 }
 
+impl Dataset {
+    /// Remove columns from the dataset.
+    ///
+    /// This is a metadata-only operation and does not remove the data from the
+    /// underlying storage. In order to remove the data, you must subsequently
+    /// call `compact_files` to rewrite the data without the removed columns and
+    /// then call `cleanup_files` to remove the old files.
```

Comment:
Most users aren't going to see this but I feel like we're going to get questions about it.  Might need to end up in a top level rst document someday.

Reply:
Oh I should add this to the Python API docs.